Appointment information sheet revisions, #3785

stgallagher opened the pull request:
### Summary

Correcting some issues with the Appointment Sheet
##### Actions
- Adding date as a passed in parameter for age calculations
- Add general consent to consent list when detected
- Date range on header handles any range of dates, as opposed to being hard set to a week
- Protect against nil and blank callers
- Removed button icon for appointment sheet link, now the link is on the event name in the scheduled activities list

stgallagher committed:
added date parameter to appointment sheet #3785

stgallagher committed:
adds general consent to consent list #3785

stgallagher committed:
made date range more flexible #3785

stgallagher committed:
updated specs and features #3785

stgallagher committed:
protecting against nils #3785

Related to a bug noticed in training where 
#event_time cause a "can't convert nil to a
string" error

stgallagher committed:
blank value protection #3785

stgallagher committed:
unnecessary javascript annotation #3785

rsutphin commented:
This is a general comment, not something that needs to be changed for this work in particular.

It's fine and good to have separate commits for separate units of work, but each separate commit should result in (roughly) working software. E.g., this series of commits changes things so that the reference date for the appointment sheet comes from outside the model. But that change is spread over four commits.

To put it another way, a commit should reflect a coherent task that the developer chose as a step in developing the feature / fixing the bug / whatever. The value in having separate commits is that they reflect the mindset of the developer has the thing was being built and can be reviewed now or in the future. The commits here seem to be divided up mostly by which kinds of components were affected (models, controllers, specs). This is not useful, but capturing your mindset and process would be.

stgallagher committed:
initialize with person and date object #3785

Have the controller manage the processing of
parameters into objects for use by the model,
instead of the model handling this task for 
itself.

stgallagher committed:
reformat too-long inline unless conditional #3785

stgallagher committed:
display none found if consent blank #3785

stgallagher committed:
consent type from code display text value #3785

Had the text coming from a hard-coded case 
statement mapping. Now it is derived from the 
display text value of the NcsCode object with 
some post-formatting via #consent_print to help
make the text more concise.

stgallagher commented:
Ready for Review

### Summary

Response to latest code review.

##### Actions
- The translation of parameter strings into source objects happens in the controller now, not in the initialization block of the model
- The consent type is derived from the display text values of their respective codes, no longer hard-coded to a case statement
- Displays "this child has no consent records" now when consents are empty strings

:sailboat:

stgallagher committed:
use relation for event type display text #3785
